llvm/lib/Transforms/Instrumentation/MemorySanitizer.cpp

```diff
@@ -3944,6 +3944,30 @@ struct MemorySanitizerVisitor : public InstVisitor<MemorySanitizerVisitor> {
     }
   }
 
+  /// Handle intrinsics by applying the intrinsic to the shadows.
+  /// The origin is approximated using setOriginForNaryOp.
+  ///
+  /// For example, this can be applied to the Arm NEON vector table intrinsics
+  /// (tbl{1,2,3,4}).
+  void handleIntrinsicByApplyingToShadow(IntrinsicInst &I, unsigned int numArgOperands) {
+    IRBuilder<> IRB(&I);
+
+    // Don't use getNumOperands() because it includes the callee
+    assert (numArgOperands == I.arg_size());
```

Comment:
Why do we need this then and don't just use `I.arg_size()`?

Reply:
It was there as a correctness check (we know that neon_tbl1 should have 2 arguments, neon_tbx1 should have 3 arguments). I will remove it.